### DISCUSSION TITLE
Switch to enable/disable map?

### DISCUSSION THREAD

**Discussion by joq3** [Tasshack/dreame-vacuum#726] [Ideas]
I know the mapping feature of this integration is resource-heavy, and I love the ability to check the map. However I wonder if it is possible to have it disabled by default, and when you need to look at the map, being able to enable it and have it processed and presented.

Or is this not how it works? If the map feature is disabled, will enabling it mid-cleaning only show the current location without any history or path?

I am thinking of disabling the map and using the current_room together with some kind of percentage of each room to explain where and how it goes.

**Comment by Tasshack** [comment]
You can disable the camera entity from Home Assistant for disabling rendering but i dont recomment constantly enabling and disabling the entity.

**Reply by joq3** [reply]
Why isn't this recommended? Can it cause database errors?

**Reply by Tasshack** [reply]
Because disabling it without reloading the integration does not disable map rendering and enabling it again causes automatically reload the integration by Home Assistant.

**Reply by Tasshack** [reply]
You can disable the map camera, enable the map data camera and use valetudo map card which does the rendering on browser.

**Reply by Tasshack** [reply]
Also you can try enabling Low Resolution Map option from configuration settings which reduces resource requirements by 4x.
